gapii/cc/protocol.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GAPII_PROTOCOL_H
+#define GAPII_PROTOCOL_H
+
+#include <string>
+
+#include <google/protobuf/io/coded_stream.h>
+
+namespace gapii {
+namespace protocol {
+
+// Maximum header size is maximum varint size plus message type size
+const uint8_t kMaxHeaderSize = 10u + 1u;
+
+enum class MessageType : uint8_t {
+  kData = 0x00u,
+  kStartTrace = 0x01u,
+  kEndTrace = 0x02u,
+  kError = 0x03u
+};
+
+inline std::string createHeader(MessageType msg_type, uint64_t msg_size = 0u) {
```

Comment:
Since even using an 8byte header is < 2% overhead in non-buffered mode. Let's make the header fixed size to simplify all the encoding/decoding code.

Reply:
No problem to go for fixed size, but the 8 byte header would come with a 4GB chunk size limitation. How to deal with that?

Reply:
The bugs in question are unrelated. They talk about particular resources, not data chunks. In this case specifically buffers are split into smaller packets during MEC capture.

Reply:
I have changed the protocol to use a fixed-size header, using 1 byte for the type and 8 bytes for the size to avoid problems with 4GB+ chunks. Splitting data would not agree with "keep things simple". :)